### DISCUSSION TITLE
How to use wails as a service

### DISCUSSION THREAD

**Discussion by KnockOutEZ** [wailsapp/wails#3376] [Q&A] [answered]
I would like to use as a service in my application. Currently im using cobra in order to start each service. (e,g `go run . start` starts web server). Just like this I want to start my wails app too using `go run . app`. How can I achieve this. Is there any way of using the wails cli commands even when the `wails.Run` is initialized in a different package other than main. Heres my commit on how im trying to distribute the code right now. https://github.com/nexentra/midgard/commit/a74d4291135033dee8594b12dfc370ac584d18d4

**Comment by leaanthony** [comment, marked as answer]
Hi 👋 I'm not sure I fully understand what you are trying to do, but if you're looking at building the wails part of your app like you do the other parts of your app then there is this guide: https://wails.io/docs/next/guides/manual-builds 
`go run` will work fine so long as you use the correct build tags and have your assets built in the right place.

**Reply by KnockOutEZ** [reply]
Thanks for the reply. I tried to build the app but its throwing this error:
<img width="924" alt="Screenshot 2024-04-07 at 5 07 28 AM" src="https://github.com/wailsapp/wails/assets/70368615/387fa9da-b168-4e37-ad08-f0c250a7d98b">

Just in case you need more context on system:
Im using latest macOS, nodejs and Go versions

**Reply by leaanthony** [reply]
I don't know if it's the right answer but you'll need cgo enabled for darwin builds

**Comment by KnockOutEZ** [comment]
Thanks for the replies. I was able to set the NewApp method as default for my application and use it as usual with wails cli commands
